### ISSUE TITLE
AVX512 triggers error in excited multideterminant calculations

### ISSUE THREAD

**Issue by anbenali** [QMCPACK/qmcpack#2781]
**Describe the bug**

The CIPSI energy for an  Excited state is : -9.97376963197278
On a laptop with Intel compiler:                  -9.974971 +/- 0.001434
On Theta with Intel compiler and AVX512: -9.881233 +/- 0.001175

**To Reproduce**

(You need the branch: MSD-ExcitedStates  in my fork to have access to the excited state)
`git https://github.com/anbenali/qmcpack.git 
 cmake -DQMC_COMPLEX=1 ..`
3. You can test with `qmcpack/test/diamondC_1x1x1-Gaussian_pp_MSD_excited/` 

**Expected behavior**
You must find same energy as sCI.
Note that GCC build on Theta generates correct energy. This is probably AVX512 aggressive optimization triggering something.  

**System:**
 - system name: Theta
 - modules loaded:
```
Currently Loaded Modulefiles:
  1) modules/3.2.11.4                                 7) pmi/5.0.16                                      13) alps/6.6.59-7.0.2.1_3.32__g872a8d62.ari         19) cray-mpich/7.7.14                               25) cray-hdf5-parallel/1.10.6.1
  2) intel/19.1.0.166                                 8) dmapp/7.1.1-7.0.2.1_2.45__g38cf134.ari          14) rca/2.2.20-7.0.2.1_2.46__g8e3fb5b.ari           20) nompirun/nompirun                               26) gcc/9.3.0
  3) craype-network-aries                             9) gni-headers/5.0.12.0-7.0.2.1_2.9__g3b1768f.ari  15) atp/3.8.1                                       21) adaptive-routing-a3                             27) miniconda-3/latest
  4) craype/2.6.5                                    10) xpmem/2.2.20-7.0.2.1_2.38__g87eb960.ari         16) perftools-base/20.06.0                          22) darshan/3.2.1                                   28) cmake/3.16.2
  5) udreg/2.3.2-7.0.2.1_2.26__g8175d3d.ari          11) job/2.2.4-7.0.2.1_2.42__g36b56f4.ari            17) PrgEnv-intel/6.0.7                              23) xalt
  6) ugni/6.0.14.0-7.0.2.1_3.39__ge78e5b0.ari        12) dvs/2.12_2.2.170-7.0.2.1_6.2__gfe7b137d         18) craype-mic-knl                                  24) cmake/3.18.0

```

**Comment by prckent** [comment]
Could it be a bug in the fork though? #2755 hasn't been updated in terms of comments in a while

If you haven't seen anything that affects in mainline, I think we need to wait for the #2755 to be completed and go through the review process.

**Comment by jtkrogel** [comment]
Does the ground state give the correct energy with and without AVX512 in mainline?

**Comment by prckent** [comment]
@anbenali Can you check the ground state energy? If it is broken we might want to delay the 3.10 release. I have the release notes up ( #2783 ), and I think this is the only concern.

**Comment by anbenali** [comment]
I will close this issue as he error was triggered by another bug I introduced and  @markdewing  found. (Order of orbitals coefficients read differently from real to complex wavefunctions and was not done properly for the case when complex code used to read real wavefunction). Bug is fixed in PR #2755
